docs/blog/2018-10-16-unstructured-data/index.md

```diff
@@ -0,0 +1,229 @@
+      component: require.resolve("./src/templates/pokemon.js"),
+      context: { pokemon },
+    })
+  })
+}
+```
+
+#### 6. Create a page [for each ability of each Pokémon](https://github.com/jlengstorf/gatsby-with-unstructured-data/blob/0a91d87b9d4d24a0e6b04b33cc271e054b7467b6/gatsby-node.js#L40).
+
+```javascript{19-26}:title=gatsby-node.js
+exports.createPages = async ({ actions: { createPage } }) => {
+  const allPokemon = await getPokemonData(["pikachu", "charizard", "squirtle"])
+
+  // Create a page that lists all Pokémon.
+  createPage({
+    path: `/`,
+    component: require.resolve("./src/templates/all-pokemon.js"),
+    context: { allPokemon },
+  })
+
+  // Create a page for each Pokémon.
+  allPokemon.forEach(pokemon => {
+    createPage({
+      path: `/pokemon/${pokemon.name}/`,
+      component: require.resolve("./src/templates/pokemon.js"),
+      context: { pokemon },
+    })
+
+    // Create a page for each ability of the current Pokémon.
+    pokemon.abilities.forEach(ability => {
+      createPage({
+        path: `/pokemon/${pokemon.name}/ability/${ability.name}/`,
+        component: require.resolve("./src/templates/ability.js"),
+        context: { pokemon, ability },
+      })
+    })
+  })
+}
+```
+
+For each type of page, we are invoking the `createPage` action, and supplying it with our desired path, React component, and data (as props on the context).
+
+> View the full source code of this example at Jason Lengstorf's ["gatsby-with-unstructured-data" repo](https://github.com/jlengstorf/gatsby-with-unstructured-data). Also check out the ["using-gatsby-data-layer" branch](https://github.com/jlengstorf/gatsby-with-unstructured-data/tree/using-gatsby-data-layer) of that repo, to compare a refactor that uses Gatsby's data layer in the same example.
+
+## When might using "unstructured data" make sense?
+
+You may find this approach useful when using Gatsby's data layer feels a bit too heavy-handed for your project scope.
+
+## The pros of using unstructured data
+
+- The approach is familiar and comfortable, especially if you’re new to GraphQL
+- There’s no intermediate step: you fetch some data, then build pages with it
+
+## The tradeoffs of foregoing Gatsby's data layer
+
+Using Gatsby's data layer provides the following benefits:
+
+- Enables you to declaratively specify what data a page component needs, alongside the page component
+- Eliminates frontend data boilerplate — no need to worry about requesting & waiting for data. Just ask for the data you need with a GraphQL query and it’ll show up when you need it
+- Pushes frontend complexity into queries — many data transformations can be done at build-time within your GraphQL queries
+- It’s the perfect data querying language for the often complex/nested data dependencies of modern applications
+- Improves performance by removing data bloat — GraphQL is a big part of why Gatsby is so fast as it enables lazy-loading the exact data in the exact form each view needs
```

Comment:
this isn't true actually — adding data via pageContext is also lazy loaded alongside graphql query results. It's more that GraphQL enables you selecting only the data you need and not whatever an API returns

Reply:
This is pulled from this section: https://www.gatsbyjs.org/docs/querying-with-graphql/#why-is-graphql-so-cool

I'll change it there, too.

Reply:
Oh hmm yeah I guess it's not really true that graphql enables lazy loading. It's a property of gatsby